### PR TITLE
Add `LocalizedViews` workload

### PR DESCRIPTION
**Pull request by dougbu**
- also react to recent `IApplicationEnvironment` changes

### CODE REVIEW

**Comment by dougbu** [comment]
:up::date:

**Comment by dougbu** [comment]
@rynowak:
> leave existing `kestrel` commands alone for muscle memory ... (Could delete most `hosting.json` files since Kestrel defaults to http://localhost:5000.)

Do we actually need these `kestrel` commands and `hosting.json` files for the non-`http://+:5000` cases?

**Comment by rynowak** [comment]
> Do we actually need these kestrel commands and hosting.json files for the non-http://+:5000 cases?

If you want to do a pass and make everything uniform be my guest. I can't think of a case we wouldn't have a command named kestrel that runs kestrel on `http://+:5000`. Other than that I don't have any strong feelings.

File: workloads/LocalizedViews/Components/CurrentUser.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+namespace BasicViews.Components
+{
+    public class CurrentUser : ViewComponent
+    {
+        private static readonly string[] Names = { "Curly", "Curly Joe", "Joe", "Larry", "Moe", "Shemp" };
+        private static int index = 0;
+
+        public string Invoke()
+        {
+            index = index++ / Names.Length;
```
Comment (rynowak) [inline workloads/LocalizedViews/Components/CurrentUser.cs]: @eilon says that `++` is really complicated and should be it's own statement so he's not confused by it.

Comment (rynowak) [inline workloads/LocalizedViews/Components/CurrentUser.cs]: I think it's fine though. He has no power here

File: workloads/LocalizedViews/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                .AddMvc()
+                .AddViewLocalization(options => options.ResourcesPath = "Resources");
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            // Default to en-CA locale and make sure nothing overrides that choice.
+            var options = new RequestLocalizationOptions();
+            options.RequestCultureProviders.Clear();
+            app.UseRequestLocalization(options, new RequestCulture("en-CA"));
```
Comment (rynowak) [inline workloads/LocalizedViews/Startup.cs]: :heart_eyes_cat:

**Comment by rynowak** [comment]
Looks like a good add. :shipit: Should we also be providing a display name for the model properties and localizing that? `Favorite Color` vs `Favourite Colour`

File: workloads/LocalizedViews/Components/CurrentUser.cs (hunk 1 above)
Comment (dougbu) [inline workloads/LocalizedViews/Components/CurrentUser.cs]: However the @Eilon is always right.

Comment (rynowak) [inline workloads/LocalizedViews/Components/CurrentUser.cs]: image

**Comment by dougbu** [comment]
> Should we also be providing a display name for the model properties and localizing that?

At best would demonstrate `[Display(Name = nameof(Resources.DisplayAttribute_Name), ResourceType = typeof(Resources))]`. MVC doesn't localize display name using the new system (hmm, can't find that issue).

Note this workload does exercise the `LanguageViewLocationExpander`. We'll see if the `IEnumerable<string>` state machine in that class shows up when nothing matches the additions, as occurs here.

**Comment by dougbu** [comment]
> If you want to do a pass and make everything uniform be my guest.

Done

**Comment by dougbu** [comment]
Commits b76a472 through 20abe9a, especially 85a4553. The other two are cleanup.

File: workloads/LocalizedViews/Components/CurrentUser.cs (hunk 1 above)
Comment (Eilon) [inline workloads/LocalizedViews/Components/CurrentUser.cs]: What's wrong with Dumbledore in that picture? Not enough midi-chlorians?

**Pull request closed without merging**